File: docker-examples/workers/go/simple_worker.go
package main

import (
	"encoding/json"
	"fmt"
	"io"
	"os"
)

// Task represents the input task structure
type Task struct {
	TaskId             string                 `json:"taskId"`
	WorkflowInstanceId string                 `json:"workflowInstanceId"`
	TaskType           string                 `json:"taskType"`
	InputData          map[string]interface{} `json:"inputData"`
}

// Result represents the output result structure
type Result struct {
	Status string                 `json:"status"`
	Output map[string]interface{} `json:"output,omitempty"`
	Logs   []string               `json:"logs,omitempty"`
	Reason string                 `json:"reason,omitempty"`
}

// Simple stdio worker example for Orkes Conductor CLI.
//
// This worker reads a task from stdin, processes it, and returns a result to stdout.
func main() {
	// Read task from stdin
	input, err := io.ReadAll(os.Stdin)
	if err != nil {
		outputError(fmt.Sprintf("Failed to read stdin: %v", err))
		return
	}

	// Parse task JSON
	var task Task
	if err := json.Unmarshal(input, &task); err != nil {
		outputError(fmt.Sprintf("Failed to parse task JSON: %v", err))
		return
	}

	// Get input parameters
	name := "World"
	if n, ok := task.InputData["name"].(string); ok && n != "" {
		name = n
	}

	// Get task metadata
	taskID := task.TaskId
	if taskID == "" {
		taskID = "unknown"
	}
	workflowID := task.WorkflowInstanceId
	if workflowID == "" {
		workflowID = "unknown"
	}
	taskType := task.TaskType
	if taskType == "" {
		taskType = "unknown"
	}

	// Process the task
	message := fmt.Sprintf("Hello, %s!", name)

	// Create result
	result := Result{
		Status: "COMPLETED",
		Output: map[string]interface{}{
			"message":    message,
			"taskId":     taskID,
			"workflowId": workflowID,
		},
		Logs: []string{
			fmt.Sprintf("Processing task %s of type %s", taskID, taskType),
			fmt.Sprintf("Workflow: %s", workflowID),
			fmt.Sprintf("Generated greeting for %s", name),
		},
	}

	// Output result to stdout
	if err := json.NewEncoder(os.Stdout).Encode(result); err != nil {
		outputError(fmt.Sprintf("Failed to encode result: %v", err))
	}
}

// outputError outputs an error result
func outputError(message string) {
	result := Result{
		Status: "FAILED",
		Reason: message,
		Logs:   []string{fmt.Sprintf("Error processing task: %s", message)},
	}
	json.NewEncoder(os.Stdout).Encode(result)
}

File: docker-examples/workers/dotnet/SimpleWorker.cs
using System;
using System.Collections.Generic;
using System.Text.Json;

/// <summary>
/// Simple stdio worker example for Orkes Conductor CLI.
///
/// This worker reads a task from stdin, processes it, and returns a result to stdout.
/// </summary>
class SimpleWorker
{
    class Task
    {
        public string? TaskId { get; set; }
        public string? WorkflowInstanceId { get; set; }
        public string? TaskType { get; set; }
        public Dictionary<string, object>? InputData { get; set; }
    }

    class Result
    {
        public string Status { get; set; } = "COMPLETED";
        public Dictionary<string, object>? Output { get; set; }
        public List<string>? Logs { get; set; }
        public string? Reason { get; set; }
    }

    static void Main()
    {
        try
        {
            // Read task from stdin
            string taskJson = Console.In.ReadToEnd();

            // Parse task JSON
            var task = JsonSerializer.Deserialize<Task>(taskJson);

            // Get input parameters
            string name = "World";
            if (task?.InputData != null && task.InputData.ContainsKey("name"))
            {
                var nameValue = task.InputData["name"];
                if (nameValue != null)
                {
                    name = nameValue.ToString() ?? "World";
                }
            }

            // Get task metadata
            string taskId = task?.TaskId ?? "unknown";
            string workflowId = task?.WorkflowInstanceId ?? "unknown";
            string taskType = task?.TaskType ?? "unknown";

            // Process the task
            string message = $"Hello, {name}!";

            // Create result
            var result = new Result
            {
                Status = "COMPLETED",
                Output = new Dictionary<string, object>
                {
                    { "message", message },
                    { "taskId", taskId },
                    { "workflowId", workflowId }
                },
                Logs = new List<string>
                {
                    $"Processing task {taskId} of type {taskType}",
                    $"Workflow: {workflowId}",
                    $"Generated greeting for {name}"
                }
            };

            // Output result to stdout
            string resultJson = JsonSerializer.Serialize(result);
            Console.WriteLine(resultJson);
        }
        catch (Exception ex)
        {
            // Return failure result on error
            var errorResult = new Result
            {
                Status = "FAILED",
                Reason = ex.Message,
                Logs = new List<string> { $"Error processing task: {ex.Message}" }
            };

            string errorJson = JsonSerializer.Serialize(errorResult);
            Console.WriteLine(errorJson);
        }
    }
}
